historical_surveillance/views.py

```diff
@@ -0,0 +1,328 @@
+# grade and gender form view, processes the and submits form data to the database
+def enrollment(request):
+    form = EnrollmentForms()
+    district_df = pd.DataFrame(District.objects.all().values())
+    districts_names = district_df['District_Name']
+    return render(request, 'enrollment.html', {'form': form, 'district_names': districts_names})
+
+
+# This function controls the total enrollment and school capacity form, processes the form data and submits
+# it to the database
+def enroll_class(request):
+    form = AggregateEnrollmentForms()
+    if request.method == 'POST':
+        form = AggregateEnrollmentForms(request.POST)
+        if form.is_valid():
+            school_form = form.cleaned_data['name_of_school']
+            academic_year_form = form.cleaned_data['academic_year']
+            district_form = form.cleaned_data['district_of_school']
+            if not AggregateEnrollment.objects.filter(name_of_school=school_form, academic_year=academic_year_form,
+                                                      district_of_school=district_form).exists():
+                form.save()
+                return redirect("/")
+        else:
+            messages.add_message(request, messages.INFO, 'Data has been entered previously')
+
+    return render(request, 'enroll_class.html', {'form': form})
+
+
+# This view controls the enrollment / Capacity views table
+def enrolled(request):
+    data = AggregateEnrollment.objects.all()
+
+    stu = {
+        "name_of_school": data
+    }
+    return render(request, 'enrolled.html', stu)
+
+#change this to get a form to select the district and pass it as a parameter to present filter the table and present
+# the data for each district
+def enrolled_trend_district_1(request):
+    error_message = None
+    district_df = pd.DataFrame(District.objects.all().values())
+    districts_names = district_df['District_Name']
+    if district_df.shape[0] > 0:
+        if request.method == 'POST':
+            district_selected = request.POST.get('district_name', False)
+            if not district_selected:
+                error_message = "Please select a chart type and school"
+            else:
+                if district_selected is not False:
+                    data = AggregateEnrollment.objects.all().filter(district_of_school=district_selected)
+                stu = {
+                    "error_message": error_message,
+                    "districts_names": districts_names,
+                    "district_selected" : district_selected,
+                    "data" : data,
+                }
+                return render(request, 'district_1.html', stu)
+        else:
+            return render(request, 'district_1.html', {"districts_names":districts_names})
+
+
+
+    else:
+        error_message = "No records found for the selected district"
+
+
+
+def district_1_trend(request):
+    error_message = None
+    graph = None
+
+    district_df = pd.DataFrame(District.objects.all().values())
```

Comment:
again here we are making a data frame just to check if there is data. I suggest just doing a simple count 

District.objects.count() > 0